lib/ansible/modules/cloud/amazon/rds_instance.py

```diff
@@ -441,7 +454,15 @@ def replicate_db_instance(module, conn):
 
 def delete_db_instance(module, conn):
     required_vars = ['instance_name']
-    valid_vars = ['snapshot', 'skip_final_snapshot']
+
+    #we have to accept but ignore variables which have defaults
+    valid_vars = ['snapshot', 'skip_final_snapshot', 'storage_type']
```

Comment:
For `storage_type`, I was thinking of removing the default from the argument_spec, and letting the AWS API handle the default. This would mean updating the `diff` check to handle storage_type though (probably using your `leave_if_null` implementation)